packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -105,6 +81,9 @@ export function TransferPanel() {
   const [destinationAddress, setDestinationAddress] = useState<
     string | undefined
   >(undefined)
+  const [allowanceL1, setAllowanceL1] = useState<BigNumber | null>(null)
```

Comment:
We could create a modified version of `useAllowance` from https://github.com/OffchainLabs/arbitrum-token-bridge/pull/733/files#diff-d12c06c3e35783c300b5de4a73d9202195e42b7c685961a60120711f8c0a0697:

```
function useTokenAllowance(erc20Address: string, provider: Provider)
```

We can use `getL1Network` and `getL2Network` from the SDK to determine if the Provider is for the L1 or the L2 network and then from there determine which gateway router to connect to etc